drivers/network/meson/ethernet.c

```diff
@@ -42,474 +38,261 @@ struct descriptor {
-    eth_dma->busmode |= DMAMAC_SRST;
-    while (eth_dma->busmode & DMAMAC_SRST);
+    eth_dma->busmode |= DMAMAC_SWRST;
+    while (eth_dma->busmode & DMAMAC_SWRST);
 
-    /* Reset removes the mac address */
-    set_mac(eth_mac, mac);
+    /* Perform flush */
+    eth_dma->opmode = FLUSHTXFIFO;
+    while (eth_dma->opmode & FLUSHTXFIFO);
 
-    eth_dma->busmode |= FIXEDBURST | PRIORXTX_41 | DMA_PBL;
-    eth_dma->opmode |= FLUSHTXFIFO | STOREFORWARD;
+    /* Reset removes the mac address */
+    eth_mac->macaddr0lo = l;
+    eth_mac->macaddr0hi = h;
 
-    eth_mac->conf |= FRAMEBURSTENABLE | DISABLERXOWN | FULLDPLXMODE;
+    eth_dma->busmode = PRIORXTX_11 | ((DMA_PBL << TX_PBL_SHFT) & TX_PBL_MASK);
+    eth_dma->opmode = STOREFORWARD;
+    eth_mac->conf = FULLDPLXMODE;
 
     eth_dma->rxdesclistaddr = hw_ring_buffer_paddr;
     eth_dma->txdesclistaddr = hw_ring_buffer_paddr + (sizeof(struct descriptor) * RX_COUNT);
 
-    eth_mac->framefilt |= 1;
+    eth_mac->framefilt |= PMSCUOUS_MODE;
 }
 
-void init_post()
+void init(void)
 {
-    /* Set up shared memory regions */
-    ring_init(&rx_ring, (ring_buffer_t *)rx_free, (ring_buffer_t *)rx_used, 0, SIZE, SIZE);
-    ring_init(&tx_ring, (ring_buffer_t *)tx_free, (ring_buffer_t *)tx_used, 0, SIZE, SIZE);
+    eth_setup();
+
+    ring_init(&rx_ring, (ring_buffer_t *)rx_free, (ring_buffer_t *)rx_used, RX_RING_SIZE_DRIV);
+    ring_init(&tx_ring, (ring_buffer_t *)tx_free, (ring_buffer_t *)tx_used, TX_RING_SIZE_DRIV);
 
-    fill_rx_bufs();
+    rx_provide();
+    tx_provide();
 
     /* Enable IRQs */
-    eth_dma->intenable |= DMA_INTR_DEFAULT_MASK;
+    eth_dma->intenable |= DMA_INTR_MASK;
+    // eth_dma->intenable |= DMA_INTR_RBU;
+    
     /* Disable uneeded GMAC irqs */
-    eth_mac->intmask |= GMAC_INT_DEFAULT_MASK;
+    eth_mac->intmask |= GMAC_INTR_MASK;
 
     /* We are ready to receive. Enable. */
-    eth_mac->conf |= RXENABLE | TXENABLE;
+    eth_mac->conf |= RX_ENABLE | TX_ENABLE;
     eth_dma->opmode |= TXSTART | RXSTART;
 
-    /* Now take away our scheduling context. Uncomment this for a passive driver. */
-    /* have_signal = true;
-    msg = seL4_MessageInfo_new(0, 0, 0, 1);
-    seL4_SetMR(0, 0);
-    signal = (MONITOR_EP); */
     microkit_irq_ack(IRQ_CH);
 }
 
-void init(void)
-{
-    //printf("%s: elf PD init function running\n", microkit_name);
-    microkit_dbg_puts("ethernet init!\n");
-
-    eth_setup();
-    init_post();
-}
-
 void notified(microkit_channel ch)
 {
     switch(ch) {
         case IRQ_CH:
-            handle_eth(eth_dma);
-            microkit_irq_ack_delayed(IRQ_CH);
+            handle_irq();
+            have_signal = true;
+            signal_msg = seL4_MessageInfo_new(IRQAckIRQ, 0, 0, 0);
+            signal_cap = (BASE_IRQ_CAP + IRQ_CH);
+            break;
+        case RX_CH:
+            rx_provide();
             break;
         case TX_CH:
-            handle_tx();
+            tx_provide();
             break;
+        // case IPBENCH_FINISH:
```

Comment:
Is this debugging code or is it needed?

Reply:
It's not needed, it is a relic of me debugging the meson ethernet driver, but it will be useful for whoever continues this work in the future. I will remove it and keep a note of it.